dipy/core/tests/test_gradients.py

```diff
@@ -412,6 +412,20 @@ def test_reorient_bvecs():
     full_affines.append(np.zeros((4, 4)))
     npt.assert_raises(ValueError, reorient_bvecs, gt_rot, full_affines)
 
+    # Shear components in the matrix need to be decomposed into rotation only,
+    # and should not lead to scaling of the bvecs
+    shear_affines = []
+    for i in np.where(~gt.b0s_mask)[0]:
+        shear_affines.append(np.array([[1, 0, 1, 0],
+                                       [0, 1, 0, 0],
+                                       [0, 0, 1, 0],
+                                       [0, 0, 0, 1]]))
+    # atol is set to 1 here to do the scaling verification here,
+    # so that the reorient_bvecs function does not throw an error itself
+    new_gt = reorient_bvecs(gt, np.array(shear_affines)[:, :3, :3], atol=1)
+    bvecs_close_to_1 = abs(vector_norm(new_gt.bvecs[~gt.b0s_mask]) - 1) <= 0.001
+    npt.assert_equal(bvecs_close_to_1, np.ones(bvecs_close_to_1.shape,
+                                               dtype=bool))
```

Comment:
I think that this is slightly clearer:

```suggestion
    npt.assert_(np.all(bvecs_close_to_1)) 
```

Reply:
I think that once we have someone take a look and verify that I did not mess this up, I think that we can merge this. Does this proposed change look OK to you, @skoudoro? In case of doubt, we can also go ahead and merge this without my suggestion.

Reply:
I agree with the suggestion and just commit it.  

When the CI's will be done, I will go ahead and merge it.

Thank you @arokem!